File: Assets/Scripts/Infrastructure/Repositories/ISaveDataRepository.cs
using Atelier.Domain.Entities;

namespace Atelier.Infrastructure.Repositories
{
    /// <summary>
    /// セーブデータの永続化を管理するリポジトリインターフェース。
    /// </summary>
    public interface ISaveDataRepository
    {
        /// <summary>
        /// セーブデータが存在するかを確認する。
        /// </summary>
        /// <returns>セーブデータが存在する場合はtrue</returns>
        bool HasSaveData();

        /// <summary>
        /// セーブデータを読み込む。
        /// </summary>
        /// <returns>読み込んだセーブデータ。存在しない場合はnull</returns>
        SaveData Load();

        /// <summary>
        /// セーブデータを保存する。
        /// </summary>
        /// <param name="data">保存するセーブデータ</param>
        void Save(SaveData data);

        /// <summary>
        /// セーブデータを削除する。
        /// </summary>
        void Delete();
    }
}

File: atelier-guild-rank-html/src/infrastructure/repository/ISaveDataRepository.ts
/**
 * セーブデータリポジトリインターフェース
 */

import { ISaveData } from '@domain/save/SaveData';

/**
 * セーブデータリポジトリインターフェース
 */
export interface ISaveDataRepository {
  /**
   * 新規セーブデータを作成
   * @param initialData 初期セーブデータ
   */
  create(initialData: ISaveData): void;

  /**
   * セーブデータを読み込み
   * @returns セーブデータ（存在しない場合はnull）
   */
  load(): ISaveData | null;

  /**
   * セーブデータを保存
   * @param data 保存するセーブデータ
   */
  save(data: ISaveData): void;

  /**
   * セーブデータを削除
   */
  delete(): void;

  /**
   * セーブデータの存在確認
   * @returns 存在する場合true
   */
  exists(): boolean;

  /**
   * セーブデータのバージョン取得
   * @returns バージョン文字列（存在しない場合はnull）
   */
  getVersion(): string | null;
}
